mesonbuild/coredata.py

```diff
@@ -964,15 +964,20 @@ def optionxform(self, optionstr: str) -> str:
         return optionstr
 
 class MachineFileParser():
-    def __init__(self, filenames: T.List[str]) -> None:
+    def __init__(self, filenames: T.List[str], sourcedir: str) -> None:
         self.parser = CmdLineFileParser()
         self.constants: T.Dict[str, T.Union[str, bool, int, T.List[str]]] = {'True': True, 'False': False}
         self.sections: T.Dict[str, T.Dict[str, T.Union[str, bool, int, T.List[str]]]] = {}
 
-        try:
-            self.parser.read(filenames)
-        except configparser.Error as e:
-            raise EnvironmentException(f'Malformed cross or native file: {e}')
+        for f in filenames:
+            with open(f, encoding='utf-8') as fp:
+                content = fp.read()
+                content = content.replace('@GLOBAL_SOURCE_ROOT@', sourcedir)
+                content = content.replace('@DIRNAME@', os.path.dirname(f))
+                try:
+                    self.parser.read_string(content)
+                except configparser.Error as e:
+                    raise EnvironmentException(f'Malformed cross or native file: {e}')
```

Comment:
I know this was just copied, but, maybe in another commit, replace "cross or native file" with "machine file"

Reply:
done